addons/mail/models/mail_message.py

```diff
@@ -940,10 +949,11 @@ def message_format(self, format_reply=True):
             message_sudo = self.browse(vals['id']).sudo().with_prefetch(self.ids)
             notifs = message_sudo.notification_ids.filtered(lambda n: n.res_partner_id)
             vals.update({
-                'needaction_partner_ids': notifs.filtered(lambda n: not n.is_read).res_partner_id.ids,
-                'history_partner_ids': notifs.filtered(lambda n: n.is_read).res_partner_id.ids,
+                'needaction_partner_ids': [{'id': p.id} for p in notifs.filtered(lambda n: not n.is_read).res_partner_id],
+                'history_partner_ids': [{'id': p.id} for p in notifs.filtered(lambda n: n.is_read).res_partner_id],
                 'is_note': message_sudo.subtype_id.id == note_id,
                 'is_discussion': message_sudo.subtype_id.id == com_id,
+                'starred_partner_ids': [{'id': p.id} for p in message_sudo.starred_partner_ids],
```

Comment:
Would rename fields to `needactionPartners`, `historyPartners`, `starredPartners`.

Reply:
Will do the rename later, in the new message format. Legacy format should use old stuff/convention.